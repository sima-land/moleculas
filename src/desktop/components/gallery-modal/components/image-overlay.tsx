import React from 'react';
import cn from 'classnames';
import styles from './image-overlay.module.scss';

export type ImageOverlayProps = React.HTMLAttributes<HTMLDivElement>;

export const ImageOverlay = ({ className, children, ...restProps }: ImageOverlayProps) => (
  <div className={cn(styles.root, className)} {...restProps}>
    {children}
  </div>
);
