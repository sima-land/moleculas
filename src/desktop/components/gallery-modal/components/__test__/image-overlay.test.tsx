import { fireEvent, render } from '@testing-library/react';
import { ImageOverlay } from '../image-overlay';

describe('ImageOverlay', () => {
  it('should renders correctly', () => {
    const spy = jest.fn();

    const { container } = render(
      <ImageOverlay className='test-class' id='test-id' onClick={spy}>
        <img src='https://images.com/2' />
      </ImageOverlay>,
    );

    expect(container.querySelectorAll('img')).toHaveLength(1);
    expect(container.querySelectorAll('#test-id')).toHaveLength(1);
    expect(container.querySelectorAll('.test-class')).toHaveLength(1);

    expect(spy).toHaveBeenCalledTimes(0);

    container.querySelectorAll('#test-id').forEach(el => {
      fireEvent.click(el);
    });

    expect(spy).toHaveBeenCalledTimes(1);
  });
});
