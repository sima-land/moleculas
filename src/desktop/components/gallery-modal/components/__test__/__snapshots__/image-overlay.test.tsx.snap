// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageOverlay should renders correctly 1`] = `
<ImageOverlay
  className="test-class"
  id="test-id"
  onClick={[MockFunction]}
>
  <div
    className="root test-class"
    id="test-id"
    onClick={[MockFunction]}
  >
    <img
      src="https://images.com/2"
    />
  </div>
</ImageOverlay>
`;
